<a href="https://colab.research.google.com/github/gitviini/atividade_estatistica/blob/main/Atividade_estat%C3%ADstica_descritiva.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Grupo:

*   Vinicius Gabriel dos Santos Braz



# Atividade Prática: Estatística Descritiva e Exploratória

Nesta atividade prática, vamos explorar os conceitos fundamentais da Estatística Descritiva e Exploratória (EDA). Utilizaremos o famoso dataset 'Tips' da biblioteca `seaborn`, que contém informações sobre gorjetas dadas em um restaurante. O objetivo é que vocês apliquem os conhecimentos teóricos na prática, manipulando dados, calculando métricas e criando visualizações para extrair insights valiosos.

### Objetivos de Aprendizagem:

*   Carregar e inspecionar um dataset.
*   Calcular e interpretar medidas de tendência central e dispersão.
*   Identificar e analisar outliers.
*   Avaliar a forma da distribuição de dados (assimetria e curtose).
*   Analisar a correlação entre variáveis numéricas.
*   Criar visualizações essenciais como histogramas e boxplots.

In [1]:
# Importando as bibliotecas necessárias
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
from scipy.stats import skew, kurtosis

# Carregando o dataset 'Tips'
df = sns.load_dataset('tips')

print("Dataset 'Tips' carregado com sucesso!")

Dataset 'Tips' carregado com sucesso!


## 1. Exploração Inicial do Dataset

Antes de mergulharmos em análises mais complexas, é crucial entender a estrutura básica dos nossos dados. A exploração inicial nos ajuda a ter uma primeira impressão das variáveis, seus tipos, valores ausentes e o tamanho geral do dataset. As funções `head()`, `info()` e `shape` são excelentes para isso.

### Exercício 1.1: Visão Geral dos Dados

Utilize os métodos `head()`, `info()` e o atributo `shape` para obter uma visão geral do dataset `df`. Observe os tipos de dados de cada coluna, a presença de valores nulos e o número de linhas e colunas.

In [37]:
df.head()

,total_bill,tip,sex,smoker,day,time,size
0,16.99,1.01,Female,No,Sun,Dinner,2
1,10.34,1.66,Male,No,Sun,Dinner,3
2,21.01,3.50,Male,No,Sun,Dinner,3
3,23.68,3.31,Male,No,Sun,Dinner,2
4,24.59,3.61,Female,No,Sun,Dinner,4


In [36]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244 entries, 0 to 243
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype   
---  ------      --------------  -----   
 0   total_bill  244 non-null    float64 
 1   tip         244 non-null    float64 
 2   sex         244 non-null    category
 3   smoker      244 non-null    category
 4   day         244 non-null    category
 5   time        244 non-null    category
 6   size        244 non-null    int64   
dtypes: category(4), float64(2), int64(1)
memory usage: 7.4 KB


In [38]:
# DIGITE SEU CÓDIGO AQUI

def showStatistic(statistic) -> pd.DataFrame:
  df_statistic = pd.DataFrame(statistic)
  return df_statistic

df.shape

(244, 7)

### Análise do Aluno:
[Escreva sua interpretação técnica aqui sobre as primeiras linhas, tipos de dados, valores não nulos e dimensões do dataset.]

### Resposta

_Por: Vinicius Gabriel_

### 1. Primeiras Linhas (`df.head()`)
A amostra inicial revela que o dataset registra transações de consumo.
- **Variáveis Numéricas:** `total_bill` e `tip` são contínuas (preços), enquanto `size` é discreta (contagem de pessoas).
- **Variáveis Categóricas:** O dataset inclui o gênero do pagador, se é fumante, o dia da semana e o período (jantar/almoço).
- **Observação:** Valores de `size` nas primeiras linhas variam entre 2 e 4.

### 2. Dimensões do Dataset (`df.shape`)
O retorno `(244, 7)` indica que temos um volume de dados moderado:
- **244 instâncias (linhas):** Cada linha representa uma mesa atendida.
- **7 atributos (colunas):** Fornecem um contexto multidimensional para cada transação.

### 3. Tipos de Dados e Integridade (`df.info()`)
Esta é a parte crucial para garantir que seus cálculos de média e desvio padrão sejam precisos:
- **Dados Limpos:** Todas as colunas possuem **244 non-null**, o que significa que não há valores ausentes (NaN). Isso dispensa a necessidade de tratamento de dados faltantes antes da análise estatística.
- **Dtypes Corretos:** * `total_bill` e `tip` como **float64** permitem cálculos decimais precisos.
    - As colunas qualitativas já estão tipadas como **category**, o que é uma boa prática de otimização de memória no Pandas.
    - `size` como **int64** confirma que não existem "meias pessoas" nas mesas.

## 2. Medidas de Tendência Central e Dispersão

As medidas de tendência central (média, mediana) nos indicam o 'centro' dos nossos dados, enquanto as medidas de dispersão (desvio padrão, variância) nos dizem o quão espalhados esses dados estão em torno desse centro. Elas são fundamentais para entender a distribuição de uma variável.

### Exercício 2.1: Cálculo de Medidas Centrais e de Dispersão

Para as colunas numéricas ('total_bill', 'tip', 'size'), calcule a média, mediana, desvio padrão e variância. Utilize o método `describe()` do pandas para uma visão rápida e depois calcule cada uma individualmente.

In [14]:
# DIGITE SEU CÓDIGO AQUI
# set columns
columns = ["total_bill", "tip", "size"]

statistic = {
    "mean": df[columns].mean(),
    "median": df[columns].median(),
    "standard deviation": df[columns].std(),
    "variance": df[columns].var()
}

# set new dataframe with statistics
print(showStatistic(statistic))

                 mean  median  standard deviation   variance
total_bill  19.785943  17.795            8.902412  79.252939
tip          2.998279   2.900            1.383638   1.914455
size         2.569672   2.000            0.951100   0.904591


### Análise do Aluno:
[Compare os valores da média e mediana para cada coluna numérica. O que esses valores sugerem sobre a simetria da distribuição? O que o desvio padrão e a variância indicam sobre a variabilidade dos dados?]

### Resposta

_Por: Vinicius Gabriel dos Santos Braz_

### 1. Simetria e Assimetria

**total_bill**: (média) 19.78 > (mediana) 17.79.

Indica assimetria positiva (direita).

- - -

**tip**: (média) 2.99 ≃ (mediana) 2.90.

Indica distribuição quase simétrica - Gorjetas tendem em torno de um valor padrão.

- - -

**size**: (média) 2.56 > (mediana) 2.00.

Indica assimetria positiva (direita).
Como variável discreta (valores inteiros) mesas com 5,6 pessoas acaba puxando o valor da média.

### 2. Variabilidade

**total_bill**:

(Desvio padrão) alto (8.90).

Isso indica uma alta variabilidade; os valores das contas variam bastante (temos desde lanches baratos até jantares caros).

(Variância) 79.25 reforça essa dispersão.

- - -

**tip**:

(Desvio padrão) é baixo (1.38).

Isso mostra que as gorjetas são relativamente consistentes. Elas não variam tanto quanto o valor total da conta, o que faz sentido, já que a gorjeta costuma ser uma porcentagem do valor total.

- - -

**size**:

(Desvio padrão) é muito baixo 0.95.

Isso indica que o tamanho dos grupos é bastante homogêneo, concentrando-se fortemente entre 2 e 4 pessoas.

## 3. Análise de Quartis e Outliers

Os quartis dividem um dataset ordenado em quatro partes iguais (Q1: 25%, Q2: 50% - mediana, Q3: 75%). Eles são cruciais para entender a dispersão e identificar possíveis outliers. Um outlier é um ponto de dado que se desvia significativamente de outras observações. Podemos identificá-los usando o Intervalo Interquartil (IQR), que é a diferença entre Q3 e Q1, e estabelecendo limites inferior e superior.

### Exercício 3.1: Cálculo Manual de Quartis e Identificação de Outliers

Para a coluna 'total_bill', calcule manualmente (sem usar `describe()` diretamente para os quartis, mas pode usar `quantile()`) os valores de Q1, Q2 (mediana), Q3 e o Intervalo Interquartil (IQR). Em seguida, calcule o Limite Inferior e Superior para identificação de outliers (Limite Inferior = Q1 - 1.5 * IQR; Limite Superior = Q3 + 1.5 * IQR). Por fim, mostre quantos outliers existem na coluna 'total_bill'.

In [ ]:
# DIGITE SEU CÓDIGO AQUI



### Análise do Aluno:
[Quantos outliers foram encontrados? O que eles representam no contexto dos dados de gorjetas? Qual a importância de identificar outliers?]

## 4. Assimetria (Skewness) e Curtose

A assimetria (skewness) mede o grau de inclinação de uma distribuição de probabilidade. Uma distribuição simétrica tem skewness zero. Valores positivos indicam uma cauda direita mais longa, enquanto valores negativos indicam uma cauda esquerda mais longa.

A curtose (kurtosis) mede o 'achatamento' ou 'pico' de uma distribuição. Uma curtose alta significa mais valores nas caudas e um pico mais nítido (leptocúrtica), enquanto uma curtose baixa indica menos valores nas caudas e um pico mais plano (platocúrtica). Uma distribuição normal tem curtose próxima de 3 (ou 0, se for calculada a curtose em excesso).

### Exercício 4.1: Cálculo e Interpretação de Assimetria e Curtose

Calcule a assimetria (skewness) e a curtose para as colunas numéricas ('total_bill', 'tip', 'size').

**Pergunta Dissertativa:** Com base nos valores calculados, descreva se a distribuição de cada variável numérica é simétrica, enviesada para a esquerda ou para a direita, e se ela é leptocúrtica, mesocúrtica ou platocúrtica. O que isso implica sobre a concentração dos dados?

In [ ]:
# DIGITE SEU CÓDIGO AQUI



### Análise do Aluno:
[Responda à pergunta dissertativa e justifique sua interpretação com base nos valores de skewness e kurtosis.]

## 5. Correlação

A correlação mede a força e a direção da relação linear entre duas variáveis numéricas. Um coeficiente de correlação próximo de 1 indica uma forte relação linear positiva, -1 indica uma forte relação linear negativa, e 0 indica nenhuma relação linear. A matriz de correlação nos permite visualizar essas relações para todos os pares de variáveis numéricas em nosso dataset.

### Exercício 5.1: Matriz de Correlação

Calcule a matriz de correlação apenas para as features numéricas do dataset (`'total_bill'`, `'tip'`, `'size'`). Em seguida, visualize essa matriz utilizando um mapa de calor (heatmap) do seaborn.

In [ ]:
# DIGITE SEU CÓDIGO AQUI



### Análise do Aluno:
[Quais pares de variáveis possuem a correlação mais forte? Essa correlação é positiva ou negativa? O que esses resultados significam no contexto do dataset 'Tips'?]

## 6. Visualização

A visualização de dados é uma parte essencial da EDA. Gráficos como histogramas e boxplots nos permitem entender rapidamente a distribuição, dispersão e a presença de outliers, muitas vezes de forma mais intuitiva do que apenas com números.

### Exercício 6.1: Histogramas com KDE

Crie histogramas com a Estimativa de Densidade Kernel (KDE) para as colunas numéricas: 'total_bill', 'tip' e 'size'. Os histogramas mostram a frequência de valores em intervalos (bins), e o KDE suaviza essa distribuição, mostrando a densidade de probabilidade.

In [ ]:
# DIGITE SEU CÓDIGO AQUI



### Análise do Aluno:
[Observe a forma de cada distribuição. Elas parecem normais? Identifique se há assimetria visualmente. O que o KDE adiciona à sua compreensão da distribuição?]

### Exercício 6.2: Boxplots

Crie boxplots para as colunas numéricas: 'total_bill', 'tip' e 'size'. Os boxplots são excelentes para visualizar a mediana, quartis, dispersão e, crucialmente, identificar outliers visualmente.

In [ ]:
# DIGITE SEU CÓDIGO AQUI



### Análise do Aluno:
[Para cada boxplot, identifique a mediana (linha central), os quartis (caixa), o alcance dos dados (bigodes) e os outliers (pontos isolados). Compare esses outliers visuais com os que você calculou no Exercício 3.1 para 'total_bill'.]

## 7. Normalização de Dados

A normalização é um processo crucial no pré-processamento de dados, especialmente quando algoritmos de Machine Learning são sensíveis à escala das features (por exemplo, modelos baseados em distância como KMeans, SVMs com kernels, ou Redes Neurais). O objetivo é ajustar os valores das features para uma escala comum, sem distorcer as diferenças nas faixas de valores ou perder informações. Existem duas técnicas comuns:

1.  **Min-Max Scaling (Normalização)**: Escala os dados para um intervalo fixo, geralmente entre 0 e 1. A fórmula é: `X_novo = (X - X_min) / (X_max - X_min)`.
2.  **StandardScaler (Padronização - Z-score)**: Transforma os dados para ter média 0 e desvio padrão 1. A fórmula é: `X_novo = (X - média) / desvio_padrão`.

### Exercício 7.1: Aplicação de Normalização Min-Max e StandardScaler

Utilize as colunas numéricas ('total_bill', 'tip', 'size') para aplicar dois métodos de normalização:

1.  **Min-Max Scaling**: Normalize estas colunas para o intervalo [0, 1]. Armazene os resultados em um novo DataFrame ou adicione colunas normalizadas ao DataFrame `df` (ex: `total_bill_minmax`).
2.  **StandardScaler (Z-score)**: Padronize as mesmas colunas para ter média 0 e desvio padrão 1. Armazene os resultados em um novo DataFrame ou adicione colunas padronizadas ao DataFrame `df` (ex: `total_bill_zscore`).

Lembre-se de importar `MinMaxScaler` e `StandardScaler` do `sklearn.preprocessing`.

In [ ]:
# DIGITE SEU CÓDIGO AQUI

from sklearn.preprocessing import MinMaxScaler, StandardScaler

# Selecionando as colunas numéricas para normalização


# 1. Min-Max Scaling



# 2. StandardScaler (Z-score)



### Análise do Aluno:
[Compare os valores originais com os valores normalizados por Min-Max e padronizados por StandardScaler. Observe as diferenças na escala e no range dos dados. Quando seria mais apropriado usar um método em vez do outro?]